File: trs-sdk-php/src/Exceptions.php
<?php

namespace TrsSdk;

class TrsException extends \RuntimeException {}

class TrsConnectionError extends TrsException {}

class TrsValidationError extends TrsException
{
    /** @var string[] */
    private array $errors;

    /** @param string[] $errors */
    public function __construct(string $message, array $errors = [])
    {
        parent::__construct($message);
        $this->errors = $errors;
    }

    /** @return string[] */
    public function errors(): array
    {
        return $this->errors;
    }
}

class TrsServerError extends TrsException {}

class TrsProtocolError extends TrsException {}

File: trs-sdk-kotlin/src/main/kotlin/dev/trs/sdk/Exceptions.kt
package dev.trs.sdk

open class TrsException(message: String, cause: Throwable? = null) : RuntimeException(message, cause)

class TrsConnectionError(message: String, cause: Throwable? = null) : TrsException(message, cause)

class TrsValidationError(message: String, val errors: List<String>) : TrsException(message)

class TrsServerError(message: String) : TrsException(message)

class TrsProtocolError(message: String) : TrsException(message)

